modules/frontend/weights/weights.go

```diff
@@ -0,0 +1,50 @@
+package weights
+
+import "github.com/grafana/tempo/pkg/traceql"
+
+// PRTODO: test weight calculation
+type WeightRequest interface {
+	SetWeight(int)
+	Weight() int
+}
+
+func TraceByID() int {
+	return 2
+}
+
+func FetchSpans(req *traceql.FetchSpansRequest) int {
+	weight := 1
+
+	if req == nil {
+		return weight
+	}
+
+	if !req.AllConditions {
+		weight++
+	}
+
+	conditions := 0
+	regexConditions := 0
+
+	for _, c := range req.Conditions {
+		if c.Op != traceql.OpNone {
+			conditions++
+		}
+		if c.Op == traceql.OpRegex || c.Op == traceql.OpNotRegex {
+			regexConditions++
+		}
+	}
+
+	if conditions > 4 { // yay, magic!
```

Comment:
A fine starting point.  I was wonder if each condition is weight++, and maybe each regex is weight+2 or some such.  It means for the queue logic that if any condition is present, we'll never consume the entire requested batch.  :thinking: